firestore/lib/src/restaurant_page.dart

```diff
@@ -0,0 +1,165 @@
+/*
+ *  Copyright 2022 Google LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import 'dart:math';
+
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:friendlyeats_flutter_firestore/src/widgets/restaurant_page_review_list.dart';
+
+import 'data/restaurant_provider.dart';
+import 'model/restaurant.dart';
+import 'model/review.dart';
+import 'widgets/dialogs/review_create.dart';
+import 'widgets/restaurant_page_details.dart';
+
+class RestaurantPage extends StatefulWidget {
+  static const route = '/restaurant';
```

Comment:
Making a note for myself that we could probably do an entire video on this route-containing `Page` widget pattern that is so common.